query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala

```diff
@@ -88,11 +88,17 @@ final case class AggregateMapReduce(aggrOp: AggregationOperator,
             limit: Int,
             sourceSchema: ResultSchema,
             paramResponse: Seq[Observable[ScalarRangeVector]] = Nil): Observable[RangeVector] = {
+
+    val needExtraKeys = QueryTimeRangeUtil.queryTimeRangeRequiresExtraKeys(
+                                    querySession.qContext.origQueryParams,
+                                    querySession.queryConfig)
+    val byLabelsFinal = if (needExtraKeys) byLabels ++ QueryTimeRangeUtil.extraKeys else byLabels
```

Comment:
Move to its own function ?

Reply:
Since it is being used in multiple places.

Reply:
Most of it has been refactored out to QueryTimeRangeUtil.*
Joins deal with `on` label and aggregations deal with `by label. Hence leaving it out of common code.